File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/timer/retrofit/TimerPresetResponseBodyConverter.java
package org.dvbviewer.controller.data.timer.retrofit;


import org.dvbviewer.controller.entities.Timer;
import org.dvbviewer.controller.io.data.TimerHandler;

import java.util.List;

import okhttp3.ResponseBody;
import retrofit2.Converter;

public class TimerPresetResponseBodyConverter implements Converter<ResponseBody, List<Timer>> {

    private final TimerHandler prefsHandler = new TimerHandler();

    @Override public List<Timer> convert(ResponseBody value) {
        try {
            return prefsHandler.parse(value.byteStream());
        } catch (RuntimeException e) {
            throw e;
        } catch (Exception e) {
            throw new RuntimeException(e);
        } finally {
            value.close();
        }
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/timer/retrofit/TimerPresetResponseBodyConverter.kt
package org.dvbviewer.controller.data.timer.retrofit


import okhttp3.ResponseBody
import org.dvbviewer.controller.data.api.handler.TimerHandler
import org.dvbviewer.controller.data.entities.Timer
import retrofit2.Converter

class TimerPresetResponseBodyConverter : Converter<ResponseBody, List<Timer>> {

    private val prefsHandler = TimerHandler()

    override fun convert(value: ResponseBody): List<Timer>? {
        try {
            return prefsHandler.parse(value.byteStream())
        } catch (e: RuntimeException) {
            throw e
        } catch (e: Exception) {
            throw RuntimeException(e)
        } finally {
            value.close()
        }
    }

}